src/overlay_94.c

```diff
@@ -0,0 +1,204 @@
+    struct UnkStruct_Overlay_94_C* unkA = unkPtr->unkc80; 
+    POKEMON* pokemon = GetPartyMonByIndex(unkPtr->unk654->party, unkPtr->partyMonIndex);
+
+    switch (unkA->unk0) {
+    case 0:
+        unkA->species = GetMonData(pokemon, MON_DATA_SPECIES, 0);
+        switch (unkA->species) {
+        case SPECIES_GIRATINA:
+            Mon_UpdateGiratinaForme(pokemon);
+            unkA->unk8 = 0x41;
+            unkA->unk10 = 0;
+            break;
+        case SPECIES_SHAYMIN:
+            Mon_UpdateShayminForme(pokemon, SHAYMIN_SKY);
+            unkA->unk8 = 0x23;
+            unkA->unk10 = 1;
+            break;
+        case SPECIES_ROTOM:
+        default:
+            GF_ASSERT(FALSE);
+            break;
+        } 
+        Pokedex_SetMonCaughtFlag(Sav2_Pokedex_get(Fsys_GetSaveDataPtr(unkPtr->unk654->fsys)), pokemon);
+        unkA->unk0++;
+        break;
+    case 1:
+    case 2:
+        unkA->unk0++;
+        break;
+    case 3:
+        ov94_021E5B04(unkPtr);
+        unkA->unk0++;
+        break;
+    case 4:
+        GX_EngineAToggleLayers(1, 1);
+        ov94_021E5BA0(unkA);
+        unkA->unk0++;
+        break;
+    case 5:
+        unkA->unk4++;
+        if(unkA->unk4 == unkA->unk8) {
+            sub_0207ECE0(unkPtr, unkPtr->partyMonIndex);
+        }
+        ov94_021E5C60();
+        if(unkA->unk4 > unkA->unk8) {
+            if(sub_020154B0(unkA->unk18) == 0) {
+                unkA->unk0++;
+            }
+        }
+        break;
+    case 6:
+        ov94_021E5B30(unkPtr);
+        unkA->unk0++;
+        break;
+    case 7:
+        sub_020720D4(pokemon);
+        unkA->unk0++;
+        break;
+    case 8:
+        if(IsCryFinished() == FALSE) {
+            unkA->unk0++;
+        }
+        break;
+    case 9:
+        STRING* str = NewString_ReadMsgData(unkPtr->msgData, 0xbc);
+        BufferBoxMonNickname(unkPtr->unk7c4, 0, Mon_GetBoxMon(pokemon));
+        StringExpandPlaceholders(unkPtr->unk7c4, unkPtr->unk7c8, str);
+        String_dtor(str);
+        sub_0207DAEC(unkPtr, ~0, 1);
+        unkA->unk0++;
+        break;
+    case 10:
+        if (sub_02020094(unkPtr->unkc64) == 0) {
+            ov94_021E5AEC(unkPtr);
+            unkPtr->unk654->unk27 = 0;
+            return TRUE;
+        }
+        break;
+    }
+    return FALSE;
+}
+
+void ov94_021E5AEC(struct UnkStruct_Overlay_94_A* unkPtr) {
+    FreeToHeap(unkPtr->unkc80);
+    unkPtr->unkc80 = 0;
+}
+
+void ov94_021E5B04(struct UnkStruct_Overlay_94_A* unkPtr) {
+    sub_0207991C(unkPtr, 0);
+    ov94_021E5B54(unkPtr->unkc80);
+    G2x_SetBlendAlpha_(0x04000050, 0, 0x3f, 0x1f, 0);
+}
+
+void ov94_021E5B30(struct UnkStruct_Overlay_94_A* unkPtr) {
+    ov94_021E5C84(unkPtr->unkc80);
+    sub_0207991C(unkPtr, 1);
+    u16* unkPtr2 = (u16*) 0x04000050;
+    unkPtr2[0] = 0;
+}
+
+void ov94_021E5B54(struct UnkStruct_Overlay_94_C* unkPtr) {
+    sub_02014DA0();
+    void* unkPtrB = AllocFromHeap(0xc, 0x4800);
```

Comment:
Heap ID should be represented in base 10. Size arg should be represented as a `sizeof` statement.